patches/ios-web_view-internal-cwv_web_view_configuration.mm.patch

```diff
@@ -0,0 +1,120 @@
+diff --git a/ios/web_view/internal/cwv_web_view_configuration.mm b/ios/web_view/internal/cwv_web_view_configuration.mm
+index b95756a291a8d2a188a4dcf12bf2420f90e6f5e2..35b6cc14cebf0db92c5c6217eaba897a45bf1230 100644
+--- a/ios/web_view/internal/cwv_web_view_configuration.mm
++++ b/ios/web_view/internal/cwv_web_view_configuration.mm
+@@ -7,12 +7,13 @@
+ #import <memory>
+ 
+ #import "base/threading/thread_restrictions.h"
+-#import "components/affiliations/core/browser/affiliation_service.h"
+ #import "components/keyed_service/core/service_access_type.h"
+ #import "components/password_manager/core/browser/leak_detection/bulk_leak_check_service_interface.h"
+ #import "components/password_manager/core/browser/password_store/password_store_interface.h"
+ #import "components/sync/service/sync_service.h"
+-#import "ios/web_view/internal/affiliations/web_view_affiliation_service_factory.h"
++#import "ios/chrome/browser/shared/model/application_context/application_context.h"
++#import "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
++#import "ios/chrome/browser/shared/model/browser_state/chrome_browser_state_manager.h"
+ #import "ios/web_view/internal/app/application_context.h"
+ #import "ios/web_view/internal/autofill/cwv_autofill_data_manager_internal.h"
+ #import "ios/web_view/internal/autofill/web_view_personal_data_manager_factory.h"
+@@ -37,7 +38,7 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ 
+ @interface CWVWebViewConfiguration () {
+   // The BrowserState for this configuration.
+-  std::unique_ptr<ios_web_view::WebViewBrowserState> _browserState;
++  ios_web_view::WebViewBrowserState* _browserState;
+ 
+   // Holds all CWVWebViews created with this class. Weak references.
+   NSHashTable* _webViews;
+@@ -76,10 +77,13 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ + (instancetype)defaultConfiguration {
+   static dispatch_once_t onceToken;
+   dispatch_once(&onceToken, ^{
+-    auto browserState = std::make_unique<ios_web_view::WebViewBrowserState>(
```

Comment:
why would we change this? It's not clear to me why chromium creates it here and why it's always on the record, but it seems odd that we would change it

Reply:
They basically recreated the API of `WKWebViewConfiguration` so to create a "private web view" you'd actually create a `nonPeristentConfiguration` and hand that to the `WKWebView`. The main difference here though is that 
 `CWVWebView` is supposed to be embeddable on its own without any reference to `//ios/chrome` & `ChromeBrowserState` which usually is what controls if the state is OTR or not. In `CWVWebView` they created a new `web::BrowserState` called `WebViewBrowserState` to house browser state and they create that browser state when you reach for a configuration.

Now of course we can't use that new browser state because we're using `ChromeBrowserState` & factories that use `ChromeBrowserState` already, so instead there's a chromium_src override that defines `WebViewBrowserState` as a subclass of `ChromeBrowserState` instead and then we have to get the actual `ChromeBrowserState` that's already been setup in BraveCoreMain.

A majority of the changes we make in chromium_src overrides for this basically come down to getting the `//ios/web_view` embedder that is supposed to be segregated from `//ios/chrome` and putting the pieces back together since we're using the `//ios/chrome` parts already.

Reply:
Sorry when I meant `nonPersistentConfiguration` I meant [nonPersistentDataStore](https://developer.apple.com/documentation/webkit/wkwebsitedatastore/1532934-nonpersistentdatastore?language=objc) which is what you'd usually pass into the configuration, but same deal basically.